src/main/java/net/neoforged/neoforge/fluids/FluidStack.java

```diff
@@ -343,6 +355,89 @@ public String toString() {
         return this.getAmount() + " " + this.getFluid();
     }
 
+    /**
+     * Builds the tooltip lines for this fluid stack, intended for use by mods
+     * rendering fluids in GUIs.
+     *
+     * <p>This mirrors the behavior of
+     * {@link ItemStack#getTooltipLines(Item.TooltipContext, Player, TooltipFlag)}
+     * as closely as possible for fluids.</p>
+     *
+     * <p>The tooltip consists of:
+     * <ul>
+     * <li>The styled hover name</li>
+     * <li>Additional lines provided by the fluid itself</li>
+     * <li>Lines added by {@link EventHooks#onFluidTooltip}</li>
+     * <li>The registry name when advanced tooltips are enabled</li>
+     * </ul>
+     * </p>
+     *
+     * <p>If tooltips are hidden via {@link TooltipDisplay} and the tooltip is not
+     * being rendered in creative mode, an empty list is returned.</p>
+     *
+     * @param context the tooltip context
+     * @param player  the player viewing the tooltip, or {@code null}
+     * @param flag    controls tooltip verbosity and advanced information
+     * @return a list of tooltip components, possibly empty
+     */
+    public List<Component> getTooltipLines(Item.TooltipContext context, @Nullable Player player, TooltipFlag flag) {
+        TooltipDisplay tooltipdisplay = this.getOrDefault(DataComponents.TOOLTIP_DISPLAY, TooltipDisplay.DEFAULT);
+        if (!flag.isCreative() && tooltipdisplay.hideTooltip()) {
+            return List.of();
+        } else {
+            Fluid fluid = getFluid();
+            List<Component> list = Lists.newArrayList();
+            list.add(this.getStyledHoverName());
+            fluid.appendTooltip(this, list, context, player, flag);
```

Comment:
would it make sense to append the potion contents component? I already append this component in one of my mods which adds a potion fluid.

On top of this are there any other vanilla components which make sense to be appended here?

Reply:
how would the wrong info be displayed?
stuff would only be appended if the component exists on the fluid stack and said component dictates a standard for how the information should be displayed.

if you want extra info or even mutate the tooltip lines the fluid tooltip event exists for that

if your doing something completely unique that is not inline with vanillas potion contents.. dont use the vanilla component, use your own and append the tooltip yourself. I see no reason why we shouldnt be appending vanillas tooltip providers (if said matching components exist on the stack)

Reply:
I think that adding FluidStack tooltips opens up some important questions, but I am hoping to avoid answering them here.

I propose that we can do this iteratively by first adding very basic tooltips. Once that's in, people can answer the questions about how to unify the representation of data like amount and common fluid components. Each of those can be done separately, and most will probably require a long discussion.

To do that well, this review should focus on whether the functionality here (or lack of) might prevent us from taking those next steps.